### DISCUSSION TITLE
Surprized about Saleor performance

### DISCUSSION THREAD

**Discussion by yacine123bain** [saleor/saleor#9209] [General]
Hi everybody,

Iam following saleor since few months now. Iam happy to see that the solution is still growing up and new features with saleor 3.
As mentionned on twitter, saleor team was involved with the developpment of this e commerce website:  https://www.lush.com/uk/en.

However, I was surprised about poor performance on https://pagespeed.web.dev/report?url=https%3A%2F%2Fwww.lush.com%2Fuk%2Fen&hl=fr, either on mobile and laptop.
Is there any explanations about that ?

Regards,

Yacine

**Comment by patrys** [comment]
Hello, Saleor Commerce does not build storefronts, and as a headless service, Saleor is entirely frontend-agnostic. We help our partners and customers best utilize the APIs Saleor provides, but those are usually invisible in PageSpeed insights. I will pass your observation to the LUSH team, and I am sure they have a good reason why the score is not perfect yet, but ultimately it's unrelated to the underlying commerce API.

**Comment by chakrit** [comment]
It is "frontend-agnostic" but not removed from frontend performance.

If GraphQL is slow to return some response, then it could impact an SSR render, for example.

I'm pretty sure it can be very very related.
